IPs/RSKIP176.md

```diff
@@ -0,0 +1,115 @@
+- *byte[]* pmtSerialized: Serialized partial merkle tree (following bitcoinj structure) that proves that the transaction belongs to the indicated block
+- *Sha256Hash* derivationArgumentsHash: A 32 bytes hash created from the derivation arguments
+- *string* userRefundBtcAddress: Base58 string representing the user BTC refund address
+- *string* LiquidityBridgeContractAddress: String representing Liquidity Bridge Contract address
+- *string* LiquidityProviderBtcAddress: Base58 string representing  the Liquidity Provider BTC refund address
+- *bool* shouldTransferToContract: Status provided by Liquidity Provider Contract to know if should transfer value to contract or not
+
+
+**Flow**:
+
+- Bridge receives a call from a contract (an alleged Liquidity Bridge Contract) and validates that the sender of that call is effectively the LBC address, that is received as parameter
+
+- Performs the same validations the Bridge does in **registerBtcTransaction** to determine if the BTC transaction is legitimate (e.g.: txAlreadyProcessed, confirmations, PMT, etc)
+
+- Verifies tx is sending funds to address derivated from Federation using the provided derivation arguments and hashing them:
+
+	`Sha256Hash(derivationArgumentsHash, userRefundAddress, LBCAddress, LPBtcAddress)`
+
+- Checks if the derivation hash was used already. If so, it will refund the user.
+
+- Verifies provided Liquidity Bridge Contract address actually belongs to a contract
+
+- Depending on the shouldTransferToContract value received as parameter from Liquidity Bridge Contract:
+
+ - If status obtained is **true**, will check if Locking Cap value is surpassed:
+    - If not surpassed:
+		- Bridge will transfer value to the Liquidity Bridge Contract
+		- will save UTXO to derived address
+		- will save UTXO in BridgeStorage (UTXO <-> Sha256Hash(Derivation Arguments))
+		- will mark tx as processed
+		- will store the UTXO for the Bridge to use, among any data concerning to the Bridge operation.
+
+	- If surpassed, will refund Liquidity Provider
+
+ - If status obtained is **false**:
+   - Refund user
+
+- Returns an integer value indicating the execution status:
+
+   ` -1: Refund user`
+
+   `-2: Refund Liquidity Provider`
+
+   `>0: Value transfered to user`
+
+
+#### Limitations
+- Transactions with depth bigger than 4320 will not be processed. That is enough depth to be able to search backwards one month worth of blocks.
+If given depth is bigger than the accepted limit, funds will be lost
+
+- If given transaction is invalid or provided address cannot be calculated from the given derivation arguments, funds will be lost
+
+
+#### Flow chart
+![registerBtcTransfer() flow](RSKIP176/flow-chart-alternative.png)
+
+
+### Federation Address Derivation
+
+The address derivation process will be carried out by creating a custom redeem script that will make a push of a 32 bytes hash created from the provided derivation arguments. That data is then dropped.
```

Comment:
It's not a "process". It's an algorithm.
"will be carried out" -> "starts with " (maybe better) ?

Try to avoid future tense "will". Make it present tense "is".